python/ray/tune/progress_reporter.py

```diff
@@ -178,6 +196,24 @@ def _progress_str(self, trials, done, *sys_info, fmt="psql", delim="\n"):
         messages.append(trial_errors_str(trials, fmt=fmt, max_rows=max_error))
         return delim.join(messages) + delim
 
+    def _infer_user_metrics(self, trials, limit=4):
+        """Try to infer the metrics to print out."""
+        if len(self._inferred_metrics) > limit:
+            return self._inferred_metrics
+        self._inferred_metrics = {}
+        for t in trials:
+            if not t.last_result:
+                continue
+            for metric, value in t.last_result.items():
+                if metric not in self.DEFAULT_COLUMNS:
+                    if metric not in AUTO_RESULT_KEYS:
+                        if type(value) in self.VALID_SUMMARY_TYPES:
+                            self._inferred_metrics[metric] = metric
+
+                if len(self._inferred_metrics) > limit:
+                    break
```

Comment:
```suggestion
                if len(self._inferred_metrics) >= limit:
                    return self._inferred_metrics
```
To break out of the outer loop. Also, we wouldn't want to exceed the limit, so `=` should be in the comparison